bulk_lookup/forms.py

```diff
@@ -0,0 +1,116 @@
+import logging
+
+from django import forms
+from django.conf import settings
+from django.utils.crypto import get_random_string
+
+import stripe
+from ukpostcodeutils.validation import is_valid_postcode
+
+from .models import OutputOption
+
+
+logger = logging.getLogger(__name__)
+
+
+class CSVForm(forms.Form):
+    original_file = forms.FileField()
+
+
+class PostcodeFieldForm(forms.Form):
+    # This is hidden by default and only shown if the CSV fails validation
+    skip_bad_rows = forms.BooleanField(
+        widget=forms.HiddenInput(),
+        required=False,
+        label="Yes, skip those bad rows"
+        )
+    postcode_field = forms.ChoiceField(required=True)
+    bad_rows = forms.IntegerField(widget=forms.HiddenInput(), required=False)
+
+    def __init__(self, *args, **kwargs):
+        self.bulk_lookup = kwargs.pop('bulk_lookup')
+        postcode_fields = self.bulk_lookup.postcode_field_choices()
+        super(PostcodeFieldForm, self).__init__(*args, **kwargs)
+        self.fields['postcode_field'].choices = postcode_fields
+
+    def clean(self):
+        cleaned_data = super(PostcodeFieldForm, self).clean()
+        postcode_field = cleaned_data.get('postcode_field')
+        skip_bad_rows = cleaned_data.get('skip_bad_rows')
+        bad_rows = 0
+        bad_row_numbers = []
+        for i, row in enumerate(self.bulk_lookup.original_file_reader()):
+            postcode = row[postcode_field].replace(" ", "")
+            if not is_valid_postcode(postcode):
+                bad_rows += 1
+                bad_row_numbers.append(str(i + 1))
```

Comment:
`bad_row_numbers` isn't ever recorded anywhere, and neither is `skip_bad_rows`, so the `process_bulk_lookups` management command will happily try and look up every row even if we know it's bad.